packages/web-components/src/utils/element-internals.ts

```diff
@@ -33,3 +33,36 @@ export function toggleState(elementInternals: ElementInternals, state: string, f
 
   elementInternals.states.delete(state);
 }
+
+/**
+ * A decorator for toggling the element internals of an element
+ *
+ * @internal
+ */
+export function toggleAttrState(target: any, attr: string): any {
+  // Stash the original method
+  const method = target[`${attr}Changed`];
+
+  // Replace the attrChanged method with a new one
+  target[`${attr}Changed`] = function (prev: string | boolean | undefined, next: string | boolean | undefined): void {
+    // Call original method
+    if (method) {
+      method.call(this, prev, next);
+    }
+
+    // Update elementInternals state for booleans
+    if (typeof next === 'boolean') {
+      toggleState(this.elementInternals, attr, next);
+      return;
+    }
+
+    // Update elementInternals state for strings
+    // NOTE: Change to `${attr}-${next}`?
```

Comment:
https://github.com/microsoft/fluentui/issues/32069

Reply:
I didn't want to overload this commit with that work but because `@toggleAttrState` rolls up 45+ `toggleState` calls into one location, that change would be a lot easier land. It would also be easier to apply and maintain consistency over time.